File: gpt-3.5-finetuned/stage2/serial_construct_vector_length_clause.f90
!T1:serial construct vector_length clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0

        REAL(8), DIMENSION(N::10, 10) :: a, b
        REAL(8) :: init

        !$acc declare create(a(N::10,10))
        !$acc declare create(b(N::10,10))

        init = 5
        a = 0
        b = 0

        !$acc update device(a, b)
        !$acc parallel
          !$acc loop vector vector_length(10)
          DO i = 1, 10
            a(i, i) = i
            b(i, i) = init
          END DO
        !$acc end parallel
        !$acc update host(a, b)
        DO i = 1, 10
          IF (a(i, i) .ne. b(i, i)) THEN
            errors = errors + 1
          END IF
        END DO

        DO j = 1, 10
          IF (b(j, j) .ne. init) THEN
            errors = errors + 1
          END IF
        END DO
        
        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
        LOGICAL :: test1
        failed = .FALSE.
        failcode = 0
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
        CALL EXIT (failcode)
      END PROGRAM

File: gpt-3.5-finetuned/stage2/serial_construct_vector_length_clause.cpp
#include "acc_testsuite.h"
#ifndef T1
//T1:serial,construct,loop,V:2.6-2.7
int test1(){
    int err = 0;
    srand(SEED);
    real_t **a = (real_t **)malloc(n * sizeof(real_t *));
    real_t **b = (real_t **)malloc(n * sizeof(real_t *));

    for(int x = 0; x < n; ++x){
        a[x] = (real_t *)malloc(n * sizeof(real_t));
        b[x] = (real_t *)malloc(n * sizeof(real_t));
    }

    for (int y = 0; y < n; ++y){
        for (int x = 0; x < n; ++x){
            a[y][x] = rand() / (real_t)(RAND_MAX / 10);
            b[y][x] = 0;
        }
    }

    #pragma acc data copyin(a[0:n][0:n]) copyout(b[0:n][0:n])
    {
        #pragma acc serial vector_length(16)
        {
            #pragma acc loop vector
            for (int y = 0; y < n; ++y){
                for (int x = 0; x < n; ++x){
                    b[y][x] = a[y][x];
                }
            }
        }
    }

    for (int y = 0; y < n; ++y){
        for (int x = 0; x < n; ++x){
            if (fabs(b[y][x] - a[y][x]) > PRECISION){
                err += 1;
            }
        }
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}